src/features/posts.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+
+type PostsType = {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsType = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    setPosts: (posts, { payload }: PayloadAction<Post[]>) => {
+      posts.items = [...payload];
```

Comment:
It's a good practice to sort or filter the payload data before setting it to the state. This can prevent potential bugs in the future.